ykrt/src/mt.rs

```diff
@@ -316,6 +311,9 @@ impl MTThread {
 /// The innards of a meta-tracer thread.
 struct MTThreadInner {
     mt: MT,
+    /// A value that uniquely identifies a thread and that is guaranteed not to overlap with
+    /// PHASE_TAG. For simplicities sake this is a pointer to a malloc'd chunk of memory.
```

Comment:
Let's distil some of the wisdom of the PR description into this comment.

Namely, we can briefly mention that most system/language provided thread IDs are non-portable and subject to change. Then it's more accurate to say "For portability, future-proofing and simplicities sake..."

Also I think "overlap" is an odd term to use here. What does it mean to overlap? I think I kind of know what you mean, but it'd be less vague if we again, took some inspiration from the PR description. For example:

> whatever ID we use it must fit into (numbits(usize) - numbits(PHASE_TAGS)) i.e. we have 62 bits for the ID on a 64-bit system.

This explains really well, especially if you mention alignment guarantees of our pointer.

Reply:
I'd prefer not to make the comment *too* long -- see if https://github.com/softdevteam/yk/pull/168/commits/e65104e0a27a693f7f6a9d7a6c916780cc35ab31 works for you.